pkg/wallet/wallet.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package wallet
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+
+	"github.com/btcsuite/btcutil/base58"
+	"golang.org/x/crypto/nacl/box"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// BaseWallet wallet implementation
+type BaseWallet struct {
+	store storage.Store
+}
+
+// key json structure include public,private and metadata
+type key struct {
+	Public   string            `json:"public"`
+	Private  *[32]byte         `json:"private"`
+	Metadata map[string]string `json:"metadata"`
+}
+
+// New return new instance of wallet implementation
+func New(storeProvider storage.Provider) (*BaseWallet, error) {
+	store, err := storeProvider.GetStoreHandle()
+	if err != nil {
+		return nil, fmt.Errorf("failed to GetStoreHandle: %w", err)
+	}
+	return &BaseWallet{store: store}, nil
+}
+
+// CreateSigningKey create a new public/private signing keypair.
+func (w *BaseWallet) CreateSigningKey(metadata map[string]string) (*KeyInfo, error) {
+	pub, priv, err := box.GenerateKey(rand.Reader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to GenerateKey: %w", err)
+	}
+	base58Pub := base58.Encode(pub[:])
+	if err := w.persistKey(&key{Public: base58Pub, Private: priv, Metadata: metadata}); err != nil {
+		return nil, err
+	}
+	return &KeyInfo{VerificationKey: base58Pub, Metadata: metadata}, nil
+}
+
+// GetSigningKey Fetch info for a signing keypair.
+func (w *BaseWallet) GetSigningKey(verKey string) (*KeyInfo, error) {
```

Comment:
I still don't understand the use for this method. This method returns the same result previously obtained from `CreateSigningKey`

Reply:
I will remove GetSigningKey

Reply:
done